ApplicationInsights/Telemetry_Client.php

```diff
@@ -158,27 +158,54 @@ public function trackMessage($message, $properties = NULL)
      */
     public function trackRequest($name, $url, $startTime, $durationInMilliseconds = 0, $httpResponseCode = 200, $isSuccessful = true, $properties = NULL, $measurements = NULL )
     {
+        $this->endRequest($this->beginRequest($name, $url, $startTime), $durationInMilliseconds, $httpResponseCode, $isSuccessful, $properties, $measurements );
+    }
+
+    /**
+     * Begin a Request_Data to the Application Insights service. This request is not sent until an @see endRequest call is made, passing this request.
+     * 
+     * @param string $name A friendly name of the request.
+     * @param string $url The url of the request.
+     * @param int $startTime The timestamp at which the request started.
+     * @return an initialized Request_Data, which can be sent by using @see endRequest
+     */    
+    public function beginRequest($name, $url, $startTime )
+    {
         $data = new Channel\Contracts\Request_Data();
         $data->setId(mt_rand());
         $data->setName($name);
         $data->setUrl($url);
         $data->setStartTime(Channel\Contracts\Utils::returnISOStringForTime($startTime));
-        $data->setResponseCode($httpResponseCode);
-        $data->setSuccess($isSuccessful);
         
-        $data->setDuration(Channel\Contracts\Utils::convertMillisecondsToTimeSpan($durationInMilliseconds));
+        return $data;
+    }
+    
+    /**
+     * Sends a Request_Data created by @see beginRequest to the Application Insights service.
+     * @param int $durationInMilliseconds The duration, in milliseconds, of the request. 
+     * @param int $httpResponseCode The response code of the request.
+     * @param bool $isSuccessful Whether or not the request was successful.
+     * @param array $properties An array of name to value pairs. Use the name as the index and any string as the value.
+     * @param array $measurements An array of name to double pairs. Use the name as the index and any double as the value.
+     */    
+    public function endRequest( Channel\Contracts\Request_Data $request, $durationInMilliseconds = 0, $httpResponseCode = 200, $isSuccessful = true, $properties = NULL, $measurements = NULL  )
+    {
+        $request->setResponseCode($httpResponseCode);
+        $request->setSuccess($isSuccessful);
+        
+        $request->setDuration(Channel\Contracts\Utils::convertMillisecondsToTimeSpan($durationInMilliseconds));
```

Comment:
i considered it, but then decided to keep the old signature intact, because the `startTime` field is getting turned into a string when set on the request, so i'd ether have to change that signature to stay a time, and have the serialization turn it into a string, or i'd have to deserialize the string back to date to do the calculation for duration.  this was much simpler.